### PR TITLE
Bump version to 2.3.4

### PR DESCRIPTION
**Pull request by EduPonz**
Signed-off-by: Eduardo Ponz Segrelles <eduardoponz@eprosima.com>

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/shapes_demo_docs_manual/26/)

**Comment by EduPonz** [comment]
@richiprosima please test this

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/shapes_demo_docs_manual/28/)

File: docs/notes/notes.rst [hunk 1]
```diff
@@ -1,13 +1,15 @@
-Version 2.3.2
+Version 2.3.4
 =============
 
-Change log:
+This release includes the following **improvements**:
 
-*   Updated to support Fast DDS 2.3.2.
+1. Update Read the Docs' python requirements
+2. Update documentation to cope with changes in Shapes Demo v2.3.4
```
Comment (MiguelCompany) [inline docs/notes/notes.rst:7]: These should be the release notes for ShapesDemo, not its documentation

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
These are the

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/shapes_demo_docs_manual/29/)

File: docs/notes/notes.rst [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
-Change log:
+This release includes the following **improvements**:
+
+1. Updated to support Fast DDS v2.3.4
+1. Updated to use Fast DDS DDS API
+1. SHM can be selected as transport
+1. Intra-process and datasharing can be enabled/disabled for Shapes Demo DomainParticipants
+1. Fast DDS Statistics module can be enabled for Shapes Demo DomainParticipants
+1. Separate participant configuration from other options
+1. Change License to GNU
```
Comment (MiguelCompany) [inline docs/notes/notes.rst:12]:
```suggestion
1. Change License to GNU GPLv3, to be compliant with Qt's license
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Comment by richiprosima** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/shapes_demo_docs_manual/30/)

**Pull request merged**